tasks/Node/src/tasks/AzureDtlCreateEnvironment/task.ts

```diff
@@ -0,0 +1,148 @@
+            parameterOverrides: tl.getInput('ParameterOverrides', false),
+            subscriptionId: tl.getEndpointDataParameter(connectedServiceName, 'SubscriptionId', true),
+            templateFile: '',
+            templateId: tl.getInput('TemplateId', true)
+        };
+    }
+
+    return inputData;
+}
+
+function showInputData(inputData: CreateOrUpdateEnvTaskInputData): void {
+    console.log('Task called with the following parameters:');
+    console.log(`  ConnectedServiceName = ${inputData.connectedServiceName}`);
+    console.log(`  LabId = ${inputData.labId}`);
+    console.log(`  EnvironmentName = ${inputData.envName}`);
+    console.log(`  TemplateId = ${inputData.templateId}`);
+    console.log(`  ParametersFile = ${inputData.parametersFile}`);
+    console.log(`  OutputTemplateVariables = ${inputData.outputTemplateVariables}`);
+    console.log(`  ExportEnvironmentTemplate = ${inputData.exportEnvTemplate}`);
+    console.log(`  ExportEnvironmentTemplateLocation = ${inputData.exportEnvTemplateLocation}`);
+    console.log(`  EnvironmentTemplateLocationVariable = ${inputData.envTemplateLocationVariable}`);
+    console.log(`  EnvironmentTemplateSasTokenVariable = ${inputData.envTemplateSasTokenVariable}`);
+}
+
+async function run(envName?: string, test?: boolean) {
+    try {
+        console.log('Starting Azure DevTest Labs Create Environment Task');
+
+        const inputData: CreateOrUpdateEnvTaskInputData = getInputData(envName, test);
+
+        const clients: TaskClients = {
+            arm: await resutil.getArmClient(inputData.subscriptionId, test),
+            dtl: await resutil.getDtlClient(inputData.subscriptionId, test)
+        };
+
+        showInputData(inputData);
+
+        await createEnvironment(clients.dtl, inputData);
+
+        const envRgId: string = tl.getVariable('environmentResourceGroupId');
+        if (envRgId) {
+            if (inputData.outputTemplateVariables) {
+                const response: ResourcesGetByIdResponse = await clients.arm.resources.getById(inputData.templateId, '2016-05-15');
+                await envutil.setOutputVariables(clients.arm, envRgId, response._response.parsedBody);
+            }
+
+            if (inputData.exportEnvTemplate) {
+                const envTemplateLocation: string = tl.getVariable(inputData.envTemplateLocationVariable);
+                const envTemplateSasToken: string = tl.getVariable(inputData.envTemplateSasTokenVariable);
+                await envutil.exportEnvironmentTemplate(inputData.exportEnvTemplateLocation, envTemplateLocation, envTemplateSasToken);
+            }
+        }
+
+        tl.setResult(tl.TaskResult.Succeeded, `Lab Environment '${inputData.envName}' was successfully created.`);
+    }
+    catch (error) {
+        console.debug(error);
+        tl.setResult(tl.TaskResult.Failed, error.message);
+    }
+}
+
+const args = require('minimist')(process.argv.slice(2));
```

Comment:
Saw this in a few places. Why are you using require for minimist instead of adding an import at the top of the file?

Reply:
In case I wanted to remove it. I was not sure, as I only used it for testing. That's all. Is there harm doing it this way?

Reply:
Why would you want to use multiple import types in the same file? I like consistency and this is inconsistent IMO everything should be import or require, not mixed. Since you are targeting ES6 you should use import.

Reply:
What is the issue? Problems with their default export? I know when using babel in webpack this wouldn't make any difference, Should be able to resolve commonJs export susing the typescript transpiler as well I would expect.